packages/pn-personafisica-webapp/cypress/e2e/Contacts.cy.ts

```diff
@@ -0,0 +1,452 @@
+    cy.get('[data-testid="loading-spinner"] > .MuiBox-root').should('not.exist');
+
+    // verify copy
+    cy.log('Verify copy');
+    cy.contains(mockData.copy.additional.title);
+    cy.contains(mockData.copy.additional.subtitle);
+
+    cy.contains(mockData.copy.additional.senderPlaceholder);
+    cy.contains(mockData.copy.additional.addrTypePlaceholder);
+    cy.contains(mockData.copy.additional.pecPlaceholder);
+    cy.get('[data-testid="Special contact add button"]').should('be.disabled');
+
+    cy.get('#sender').click();
+    cy.get(`[data-value="${mockData.data.additional.sender}"]`).click();
+    cy.get('#addressType').click();
+    cy.get('[data-value="EMAIL"]').click();
+    cy.contains(mockData.copy.additional.mailPlaceholder);
+
+    cy.get('#addressType').click();
+    cy.get('[data-value="PEC"]').click();
+
+    // enter an invalid pec and verify the error message
+    cy.get('#s_pec').type(mockData.data.additional.invalid);
+    cy.get('#s_pec-helper-text')
+      .should('be.visible')
+      .should('have.text', mockData.copy.additional.inputInvalidMessage);
+
+    cy.get('[data-testid="Special contact add button"]').should('be.disabled');
+
+    // enter a valid pec and confirm
+    cy.get('#s_pec').clear().type(mockData.data.additional.valid);
+
+    cy.intercept(
+      'POST',
+      `${LEGAL_CONTACT(mockData.data.additional.sender, LegalChannelType.PEC)}`,
+      {
+        statusCode: 200,
+        fixture: '',
+      }
+    ).as('addPec');
+
+    cy.get('[data-testid="Special contact add button"]').should('be.enabled').click();
+
+    cy.get('[data-testid="code confirm button"]').should('be.disabled');
+    cy.get('[data-testid="code cancel button"]').should('be.enabled');
+
+    // insert an invalid code
+    cy.intercept(
+      'POST',
+      `${LEGAL_CONTACT(mockData.data.additional.sender, LegalChannelType.PEC)}`,
+      {
+        statusCode: 422,
+        fixture: 'contacts/invalid-code-response',
+      }
+    ).as('sendInvalidCode');
+
+    cy.get('[data-testid="code input (0)"]').type(mockData.data.codes.invalid, { delay: 100 });
+    cy.get('[data-testid="code confirm button"]').click();
+
+    // verify the request is properly formatted
+    cy.wait('@sendInvalidCode').its('request.body').should('deep.equal', {
+      value: mockData.data.additional.valid,
+      verificationCode: mockData.data.codes.invalid,
+    });
+
+    cy.get('[data-testid="errorAlert"]').should('be.visible');
+
+    // insert a valid code
+    cy.intercept(
+      'POST',
+      `${LEGAL_CONTACT(mockData.data.additional.sender, LegalChannelType.PEC)}`,
+      {
+        statusCode: 204,
+        fixture: '',
+      }
+    ).as('addPec');
+
+    cy.get('[data-testid="code input (0)"]').type(mockData.data.codes.valid, { delay: 100 });
+    cy.get('[data-testid="code confirm button"]').click();
+
+    cy.wait('@addPec');
+    cy.contains(mockData.copy.additional.successMessage);
+    cy.contains(mockData.data.additional.valid);
+  });
```

Comment:
In the confluence page about contacts test there is missing another test here. You have to add another test contact where address type is phone.

Reply:
Oh you're right... I skipped it! I'm gonna fix right now!